### ISSUE TITLE
LL Publish RSS Feed?

### ISSUE THREAD

**Issue by ethanopp** [DobyTang/LazyLibrarian#1430]
I'm aware LL can consume RSS feeds for building a list of books to download, but is it able to distribute through an RSS feed?

I'm looking to start downloading news (through magazines?) but curious how other people distribute the content.

I would prefer if I could just connect an app like flipboard to an RSS feed that LL creates rather than having to go in every day and manually download the next issue to my mobile, and then delete if after its read.

How are others using LL to read magazines?

**Comment by philborman** [comment]
Personally I use the coverwall page of recent downloads, with the browser set to automatically open pdf files in a pdf viewer.

Generating an rss feed from recent downloads is the sort of thing that could be easily done as part of the notification system. Not sure how you would go about serving the feed to clients though.

**Comment by ethanopp** [comment]
If the feed can be generated by a custom notification, couldn't we use the LL site itself to serve that feed?

**Comment by philborman** [comment]
Well, turns out serving the feed is surprisingly simple. Found a nice little python library to generate the xml from the database entries. Working well from the lazylibrarian api so you can get at it with a browser. The tricky bit is serving the content, ie you have a feed containing the most recently downloaded eBooks, it shows the book title/author/description. If you want to be able to download the book from the feed via a link it gets more tricky.

**Comment by ethanopp** [comment]
Ah I see... turns out the feed I was thinking of is going to be different from what that script would probably generate anyway... The ideal script would have a different feed item for each article in a news source (i.e. http://www.wsj.com/xml/rss/3_7041.xml), however after looking at how the magazines are downloaded, it is 1 pdf per day (all articles in 1 pdf) so I'm not sure the RSS feed would be able to break it out by article anyway...

**Comment by philborman** [comment]
Yes we can't really split the pdf file, except extracting individual pages which probably isn't what you would want.
Link to file is do-able in a browser (we can just  _serve_ the file from lazylibrarian the same way we open books) the problem is if there is more than one file like an audiobook, or more than one format of ebook. Don't know whether we should zip them up? Or for an ebook just serve the mobi?  On the rss feed we don't know who the user is and what they might want. In the lazylibrarian browser interface we can ask you to choose, can't do that from rss.

This is what I've got so far in feedreader...
![screenshot from 2018-05-21 16-36-51](https://user-images.githubusercontent.com/12158777/40313171-3ddeeb50-5d15-11e8-8b12-9504b8b85f2b.png)

**Comment by ethanopp** [comment]
How about generating different feeds for each file type? Depending on the device end user is using, they can choose to use the '.mobi' RSS feed, or the '.pdf' feed?
